src/utils/logger.js

```diff
@@ -125,16 +113,13 @@ export class Logger {
 
   logError(value: Value, message: string) {
     let loc = value.expressionLocation;
-    if (loc) {
-      let locString = `${loc.start.line}:${loc.start.column + 1}`;
-      if (loc.source) locString = `${loc.source}:${locString}`;
-      message = `${message}\nat: ${locString}`;
-    } else if (value.intrinsicName) {
+    if (value.intrinsicName) {
       message = `${message}\nintrinsic name: ${value.intrinsicName}`;
     }
 
-    console.error(message);
     this._hasErrors = true;
+    let error = new CompilerDiagnostic(message, loc, "PP9000", "RecoverableError");
```

Comment:
New error codes should be documented int the wiki. Also the range is significant. Any errors that arise because of limitations of the abstract interpreter should be in the range 0 - 999. 

Since every wiki page should be specific to the root cause, it does not do to use a single error code for every error that is reported via logError. The quick fix is to pass in the error code as a parameter.

A better fix might be to just get rid of logError.

Reply:
This is the catch all for all serializer errors. I did create a wiki page.

Reply:
And yes, the logError thing probably should be refactored at some point. This is a minimal change that fixes error reporting in general (instead of causing an internal invariant).

Reply:
There is also a wiki page that lists all known errors: https://github.com/facebook/prepack/wiki/Prepack-diagnostics